VTKh breaking when compiled with hipcc

On Crusher, building Genten requires Ascent to be built with hipcc and this causes some build errors between VTKh and ROCm. 

This is not an error specific to Genten and happens whenever Ascent is built with hipcc, it just so happens that Ascent+Genten cannot be built with amdclang (whereas Ascent alone can be). 

I'm going to post one of the errors and then post all the errors further below. Side scrolling on a wall of text is a pain so I wanted to include a smaller chunk that's easier to look over.

Using hipcc results in 12 errors of the following flavor: 
```
[ 21%] Building CXX object libs/vtkh/compositing/CMakeFiles/vtkh_compositing_mpi.dir/PartialCompositor.cpp.o
cd /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/build/libs/vtkh/compositing && /opt/rocm-5.2.0/bin/..//bin/hipcc -DASCENT_EXPORTS_FLAG -DVTKH_PARALLEL -DVTKMDIY_MPI_AS_LIB -DVTKMDIY_NO_THREADS -Dvtkh_compositing_mpi_EXPORTS -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/thirdparty_builtin/lodepng -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/thirdparty_builtin/tiny_obj -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/ascent -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/build/ascent -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/ascent/utils -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/ascent/runtimes -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/ascent/hola -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/flow -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/build/flow -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/flow/filters -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/ascent/runtimes/flow_filters -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/rover -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/build/libs -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/internal/diy/include -isystem /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/vtk-m-v1.8.0/install/include/vtkm-1.8/vtkm/thirdparty/lcl/vtkmlcl -isystem /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/vtk-m-v1.8.0/install/include/vtkm-1.8/vtkm/thirdparty/diy/vtkmdiy/include -isystem /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/vtk-m-v1.8.0/install/include/vtkm-1.8 -isystem /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/kokkos-3.6.01/install/include -isystem /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/vtk-m-v1.8.0/install/include/vtkm-1.8/vtkm/thirdparty/optionparser -isystem /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/vtk-m-v1.8.0/install/include/vtkm-1.8/vtkm/thirdparty/diy -isystem /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/vtk-m-v1.8.0/install/include/vtkm-1.8/vtkm/thirdparty/diy/vtkmdiy/include/vtkmdiy/mpi -I/opt/rocm-5.2.0/include -Wno-pass-failed -I/opt/cray/pe/mpich/8.1.16/ofi/gnu/9.1/include -D__HIP_PLATFORM_AMD__    -O3 -DNDEBUG -fPIC -fvisibility=hidden -ffunction-sections -std=c++14 -MD -MT libs/vtkh/compositing/CMakeFiles/vtkh_compositing_mpi.dir/PartialCompositor.cpp.o -MF CMakeFiles/vtkh_compositing_mpi.dir/PartialCompositor.cpp.o.d -o CMakeFiles/vtkh_compositing_mpi.dir/PartialCompositor.cpp.o -c /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/PartialCompositor.cppIn file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/PartialCompositor.cpp:49:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:45:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_blocks.hpp:45:
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp:176:60: error: reference to __device__ function 'operator delete' in __host__ function
                           QueuePolicy*         q_policy = new QueueSizePolicy(4096)): //!< policy for managing message queues specifies maximum size of message queues to keep in memory
                                                           ^
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:166:3: note: called by 'redistribute<vtkh::VolumePartial<float>>'
  redistribute_detail<AddBlock<VolumeBlock<float>>>(partials,
  ^
/opt/rocm-5.2.0/llvm/lib/clang/14.0.0/include/cuda_wrappers/new:63:24: note: 'operator delete' declared here
__device__ inline void operator delete(void* ptr) CUDA_NOEXCEPT {
                       ^
```


Here are all the errors: 
```

[ 21%] Building CXX object libs/vtkh/compositing/CMakeFiles/vtkh_compositing_mpi.dir/PartialCompositor.cpp.o
cd /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/build/libs/vtkh/compositing && /opt/rocm-5.2.0/bin/..//bin/hipcc -DASCENT_EXPORTS_FLAG -DVTKH_PARALLEL -DVTKMDIY_MPI_AS_LIB -DVTKMDIY_NO_THREADS -Dvtkh_compositing_mpi_EXPORTS -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/thirdparty_builtin/lodepng -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/thirdparty_builtin/tiny_obj -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/ascent -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/build/ascent -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/ascent/utils -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/ascent/runtimes -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/ascent/hola -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/flow -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/build/flow -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/flow/filters -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/ascent/runtimes/flow_filters -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/rover -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/build/libs -I/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/internal/diy/include -isystem /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/vtk-m-v1.8.0/install/include/vtkm-1.8/vtkm/thirdparty/lcl/vtkmlcl -isystem /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/vtk-m-v1.8.0/install/include/vtkm-1.8/vtkm/thirdparty/diy/vtkmdiy/include -isystem /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/vtk-m-v1.8.0/install/include/vtkm-1.8 -isystem /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/kokkos-3.6.01/install/include -isystem /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/vtk-m-v1.8.0/install/include/vtkm-1.8/vtkm/thirdparty/optionparser -isystem /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/vtk-m-v1.8.0/install/include/vtkm-1.8/vtkm/thirdparty/diy -isystem /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/vtk-m-v1.8.0/install/include/vtkm-1.8/vtkm/thirdparty/diy/vtkmdiy/include/vtkmdiy/mpi -I/opt/rocm-5.2.0/include -Wno-pass-failed -I/opt/cray/pe/mpich/8.1.16/ofi/gnu/9.1/include -D__HIP_PLATFORM_AMD__    -O3 -DNDEBUG -fPIC -fvisibility=hidden -ffunction-sections -std=c++14 -MD -MT libs/vtkh/compositing/CMakeFiles/vtkh_compositing_mpi.dir/PartialCompositor.cpp.o -MF CMakeFiles/vtkh_compositing_mpi.dir/PartialCompositor.cpp.o.d -o CMakeFiles/vtkh_compositing_mpi.dir/PartialCompositor.cpp.o -c /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/PartialCompositor.cppIn file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/PartialCompositor.cpp:49:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:45:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_blocks.hpp:45:
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp:176:60: error: reference to __device__ function 'operator delete' in __host__ function
                           QueuePolicy*         q_policy = new QueueSizePolicy(4096)): //!< policy for managing message queues specifies maximum size of message queues to keep in memory
                                                           ^
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:166:3: note: called by 'redistribute<vtkh::VolumePartial<float>>'
  redistribute_detail<AddBlock<VolumeBlock<float>>>(partials,
  ^
/opt/rocm-5.2.0/llvm/lib/clang/14.0.0/include/cuda_wrappers/new:63:24: note: 'operator delete' declared here
__device__ inline void operator delete(void* ptr) CUDA_NOEXCEPT {
                       ^
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/PartialCompositor.cpp:49:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:45:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_blocks.hpp:45:
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp:176:60: error: reference to __device__ function 'operator delete' in __host__ function
                           QueuePolicy*         q_policy = new QueueSizePolicy(4096)): //!< policy for managing message queues specifies maximum size of message queues to keep in memory
                                                           ^
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:178:3: note: called by 'redistribute<vtkh::VolumePartial<double>>'
  redistribute_detail<AddBlock<VolumeBlock<double>>>(partials,
  ^
/opt/rocm-5.2.0/llvm/lib/clang/14.0.0/include/cuda_wrappers/new:63:24: note: 'operator delete' declared here
__device__ inline void operator delete(void* ptr) CUDA_NOEXCEPT {
                       ^
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/PartialCompositor.cpp:49:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:45:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_blocks.hpp:45:
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp:176:60: error: reference to __device__ function 'operator delete' in __host__ function
                           QueuePolicy*         q_policy = new QueueSizePolicy(4096)): //!< policy for managing message queues specifies maximum size of message queues to keep in memory
                                                           ^
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:191:3: note: called by 'redistribute<vtkh::AbsorptionPartial<double>>'
  redistribute_detail<AddBlock<AbsorptionBlock<double>>>(partials,
  ^
/opt/rocm-5.2.0/llvm/lib/clang/14.0.0/include/cuda_wrappers/new:63:24: note: 'operator delete' declared here
__device__ inline void operator delete(void* ptr) CUDA_NOEXCEPT {
                       ^
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/PartialCompositor.cpp:49:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:45:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_blocks.hpp:45:
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp:176:60: error: reference to __device__ function 'operator delete' in __host__ function
                           QueuePolicy*         q_policy = new QueueSizePolicy(4096)): //!< policy for managing message queues specifies maximum size of message queues to keep in memory
                                                           ^
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:203:3: note: called by 'redistribute<vtkh::AbsorptionPartial<float>>'
  redistribute_detail<AddBlock<AbsorptionBlock<float>>>(partials,
  ^
/opt/rocm-5.2.0/llvm/lib/clang/14.0.0/include/cuda_wrappers/new:63:24: note: 'operator delete' declared here
__device__ inline void operator delete(void* ptr) CUDA_NOEXCEPT {
                       ^
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/PartialCompositor.cpp:49:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:45:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_blocks.hpp:45:
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp:176:60: error: reference to __device__ function 'operator delete' in __host__ function
                           QueuePolicy*         q_policy = new QueueSizePolicy(4096)): //!< policy for managing message queues specifies maximum size of message queues to keep in memory
                                                           ^
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:216:3: note: called by 'redistribute<vtkh::EmissionPartial<double>>'
  redistribute_detail<AddBlock<EmissionBlock<double>>>(partials,
  ^
/opt/rocm-5.2.0/llvm/lib/clang/14.0.0/include/cuda_wrappers/new:63:24: note: 'operator delete' declared here
__device__ inline void operator delete(void* ptr) CUDA_NOEXCEPT {
                       ^
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/PartialCompositor.cpp:49:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:45:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_blocks.hpp:45:
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp:176:60: error: reference to __device__ function 'operator delete' in __host__ function
                           QueuePolicy*         q_policy = new QueueSizePolicy(4096)): //!< policy for managing message queues specifies maximum size of message queues to keep in memory
                                                           ^
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:228:3: note: called by 'redistribute<vtkh::EmissionPartial<float>>'
  redistribute_detail<AddBlock<EmissionBlock<float>>>(partials,
  ^
/opt/rocm-5.2.0/llvm/lib/clang/14.0.0/include/cuda_wrappers/new:63:24: note: 'operator delete' declared here
__device__ inline void operator delete(void* ptr) CUDA_NOEXCEPT {
                       ^
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/PartialCompositor.cpp:49:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:45:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_blocks.hpp:45:
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp:176:60: error: reference to __device__ function 'operator delete' in __host__ function
                           QueuePolicy*         q_policy = new QueueSizePolicy(4096)): //!< policy for managing message queues specifies maximum size of message queues to keep in memory
                                                           ^
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_collect.hpp:151:3: note: called by 'collect<vtkh::VolumePartial<float>>'
  collect_detail<AddBlock<VolumeBlock<float>>>(partials, comm);
  ^
/opt/rocm-5.2.0/llvm/lib/clang/14.0.0/include/cuda_wrappers/new:63:24: note: 'operator delete' declared here
__device__ inline void operator delete(void* ptr) CUDA_NOEXCEPT {
                       ^
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/PartialCompositor.cpp:49:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:45:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_blocks.hpp:45:
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp:176:60: error: reference to __device__ function 'operator delete' in __host__ function
                           QueuePolicy*         q_policy = new QueueSizePolicy(4096)): //!< policy for managing message queues specifies maximum size of message queues to keep in memory
                                                           ^
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_collect.hpp:158:3: note: called by 'collect<vtkh::VolumePartial<double>>'
  collect_detail<AddBlock<VolumeBlock<double>>>(partials, comm);
  ^
/opt/rocm-5.2.0/llvm/lib/clang/14.0.0/include/cuda_wrappers/new:63:24: note: 'operator delete' declared here
__device__ inline void operator delete(void* ptr) CUDA_NOEXCEPT {
                       ^
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/PartialCompositor.cpp:49:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:45:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_blocks.hpp:45:
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp:176:60: error: reference to __device__ function 'operator delete' in __host__ function
                           QueuePolicy*         q_policy = new QueueSizePolicy(4096)): //!< policy for managing message queues specifies maximum size of message queues to keep in memory
                                                           ^
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_collect.hpp:165:3: note: called by 'collect<vtkh::AbsorptionPartial<double>>'
  collect_detail<AddBlock<AbsorptionBlock<double>>>(partials, comm);
  ^
/opt/rocm-5.2.0/llvm/lib/clang/14.0.0/include/cuda_wrappers/new:63:24: note: 'operator delete' declared here
__device__ inline void operator delete(void* ptr) CUDA_NOEXCEPT {
                       ^
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/PartialCompositor.cpp:49:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:45:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_blocks.hpp:45:
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp:176:60: error: reference to __device__ function 'operator delete' in __host__ function
                           QueuePolicy*         q_policy = new QueueSizePolicy(4096)): //!< policy for managing message queues specifies maximum size of message queues to keep in memory
                                                           ^
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_collect.hpp:172:3: note: called by 'collect<vtkh::AbsorptionPartial<float>>'
  collect_detail<AddBlock<AbsorptionBlock<float>>>(partials, comm);
  ^
/opt/rocm-5.2.0/llvm/lib/clang/14.0.0/include/cuda_wrappers/new:63:24: note: 'operator delete' declared here
__device__ inline void operator delete(void* ptr) CUDA_NOEXCEPT {
                       ^
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/PartialCompositor.cpp:49:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:45:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_blocks.hpp:45:
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp:176:60: error: reference to __device__ function 'operator delete' in __host__ function
                           QueuePolicy*         q_policy = new QueueSizePolicy(4096)): //!< policy for managing message queues specifies maximum size of message queues to keep in memory
                                                           ^
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_collect.hpp:179:3: note: called by 'collect<vtkh::EmissionPartial<double>>'
  collect_detail<AddBlock<EmissionBlock<double>>>(partials, comm);
  ^
/opt/rocm-5.2.0/llvm/lib/clang/14.0.0/include/cuda_wrappers/new:63:24: note: 'operator delete' declared here
__device__ inline void operator delete(void* ptr) CUDA_NOEXCEPT {
                       ^
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/PartialCompositor.cpp:49:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_redistribute.hpp:45:
In file included from /gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_blocks.hpp:45:
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp:176:60: error: reference to __device__ function 'operator delete' in __host__ function
                           QueuePolicy*         q_policy = new QueueSizePolicy(4096)): //!< policy for managing message queues specifies maximum size of message queues to keep in memory
                                                           ^
/gpfs/alpine/scratch/nic8504/csc340/2022_08_kokkos_genten_ascent/ascent-crusher-build-scripts/ascent-v0.9.0-pre/src/libs/vtkh/compositing/vtkh_diy_partial_collect.hpp:186:3: note: called by 'collect<vtkh::EmissionPartial<float>>'
  collect_detail<AddBlock<EmissionBlock<float>>>(partials, comm);
  ^
/opt/rocm-5.2.0/llvm/lib/clang/14.0.0/include/cuda_wrappers/new:63:24: note: 'operator delete' declared here
__device__ inline void operator delete(void* ptr) CUDA_NOEXCEPT {
                       ^
12 errors generated when compiling for host.
```

--- a/src/libs/apcomp/internal/diy/include/diy/master.hpp
+++ b/src/libs/apcomp/internal/diy/include/diy/master.hpp
@@ -86,6 +86,8 @@ namespace apcompdiy
       struct QueuePolicy
       {
                 QueuePolicy(size_t sz): size(sz)          {}
+                QueuePolicy(const QueuePolicy &p) size(p.size) {}
+               ~QueuePolicy() {}
         bool    unload_incoming(const Master&, int, int, size_t sz) const         { return sz > size; }
         bool    unload_outgoing(const Master& master, int from, size_t sz) const  { return sz > size*master.outgoing_count(from); }
 

--- a/src/libs/apcomp/internal/diy/include/diy/master.hpp
+++ b/src/libs/apcomp/internal/diy/include/diy/master.hpp
@@ -72,22 +72,26 @@ namespace apcompdiy
       template<class Block>
       using ICallback = std::function<bool(Block*, const ProxyWithLink&)>;
 
-      struct QueuePolicy
-      {
-        virtual bool    unload_incoming(const Master& master, int from, int to, size_t size) const  =0;
-        virtual bool    unload_outgoing(const Master& master, int from, size_t size) const          =0;
-        virtual         ~QueuePolicy() {}
-      };
+      // BEGIN ASCENT EDIT //
+      // Only use a single policy, avoid new of QueuePolicy
+      // due to issues with hipcc and def of delete operator
+      // struct QueuePolicy
+      // {
+      //   virtual bool    unload_incoming(const Master& master, int from, int to, size_t size) const  =0;
+      //   virtual bool    unload_outgoing(const Master& master, int from, size_t size) const          =0;
+      //   virtual         ~QueuePolicy() {}
+      // };
 
       //! Move queues out of core if their size exceeds a parameter given in the constructor
-      struct QueueSizePolicy: public QueuePolicy
+      struct QueuePolicy
       {
-                QueueSizePolicy(size_t sz): size(sz)          {}
+                QueuePolicy(size_t sz): size(sz)          {}
         bool    unload_incoming(const Master&, int, int, size_t sz) const         { return sz > size; }
         bool    unload_outgoing(const Master& master, int from, size_t sz) const  { return sz > size*master.outgoing_count(from); }
 
         size_t  size;
       };
+      // END ASCENT EDIT //
 
       // forward declarations, defined in detail/master/communication.hpp
       struct MessageInfo;
@@ -160,7 +164,9 @@ namespace apcompdiy
                            ExternalStorage*     storage   = 0,  //!< storage object (path, method, etc.) for storing temporary blocks being shuffled in/out of core
                            SaveBlock            save      = 0,  //!< block save function; master manages saving if save != 0
                            LoadBlock            load_     = 0,  //!< block load function; master manages loading if load != 0
-                           QueuePolicy*         q_policy  = new QueueSizePolicy(4096)); //!< policy for managing message queues specifies maximum size of message queues to keep in memory
+                           // BEGIN ASCENT EDIT //
+                           QueuePolicy          q_policy  = QueuePolicy(4096)); //!< policy for managing message queues specifies maximum size of message queues to keep in memory
+                           // END ASCENT EDIT //
       inline        ~Master();
 
       inline void   clear();
@@ -319,8 +325,9 @@ namespace apcompdiy
       Collection            blocks_;
       std::vector<int>      gids_;
       std::map<int, int>    lids_;
-
-      QueuePolicy*          queue_policy_;
+      // BEGIN ASCENT EDIT //
+      QueuePolicy           queue_policy_;
+      // END ASCENT EDIT //
 
       int                   limit_;
       int                   threads_;
@@ -369,7 +376,7 @@ Master(mpi::communicator    comm,
        ExternalStorage*     storage,
        SaveBlock            save,
        LoadBlock            load_,
-       QueuePolicy*         q_policy):
+       QueuePolicy          q_policy):
   blocks_(create_, destroy_, storage, save, load_),
   queue_policy_(q_policy),
   limit_(limit__),
@@ -388,7 +395,9 @@ apcompdiy::Master::
 {
     set_immediate(true);
     clear();
-    delete queue_policy_;
+    // BEGIN ASCENT EDIT //
+    //delete queue_policy_;
+    // END ASCENT EDIT //
 }
 
 void
@@ -437,7 +446,8 @@ unload_incoming(int gid__)
     for (InQueueRecords::iterator it = in_qrs.records.begin(); it != in_qrs.records.end(); ++it)
     {
       QueueRecord& qr = it->second;
-      if (queue_policy_->unload_incoming(*this, it->first, gid__, qr.size))
+      // ASCENT EDIT (queue_policy_ no longer a pointer)//
+      if (queue_policy_.unload_incoming(*this, it->first, gid__, qr.size))
       {
         log->debug("Unloading queue: {} <- {}", gid__, it->first);
         qr.external = storage_->put(in_qrs.queues[it->first]);
@@ -463,7 +473,8 @@ unload_outgoing(int gid__)
     out_queues_size += Serialization<MemoryBuffer>::size(rec.second);   // buffer contents
     ++count;
   }
-  if (queue_policy_->unload_outgoing(*this, gid__, out_queues_size - sizeof(size_t)))
+  // ASCENT EDIT (queue_policy_ no longer a pointer)//
+  if (queue_policy_.unload_outgoing(*this, gid__, out_queues_size - sizeof(size_t)))
   {
       log->debug("Unloading outgoing queues: {} -> ...; size = {}\n", gid__, out_queues_size);
       MemoryBuffer  bb;     bb.reserve(out_queues_size);
@@ -477,8 +488,9 @@ unload_outgoing(int gid__)
         auto& buffer = it->second;
         if (bid.proc == comm_.rank())
         {
+          // ASCENT EDIT (queue_policy_ no longer a pointer)//
           // treat as incoming
-          if (queue_policy_->unload_incoming(*this, gid__, bid.gid, buffer.size()))
+          if (queue_policy_.unload_incoming(*this, gid__, bid.gid, buffer.size()))
           {
             QueueRecord& qr = out_qr.external_local[bid];
             qr.size     = buffer.size();
@@ -1004,7 +1016,8 @@ send_same_rank(int from, int to, MemoryBuffer& bb, IExchangeInfo* iexchange)
     {
         log->debug("Unloading outgoing directly as incoming: {} <- {}", to, from);
         in_qr.size = bb.size();
-        if (queue_policy_->unload_incoming(*this, from, to, in_qr.size))
+        // ASCENT EDIT (queue_policy_ no longer a pointer)//
+        if (queue_policy_.unload_incoming(*this, from, to, in_qr.size))
             in_qr.external = storage_->put(bb);
         else
         {
@@ -1157,9 +1170,9 @@ check_incoming_queues(IExchangeInfo* iexchange)
         {
             assert(ir.info.round >= exchange_round_);
             IncomingRound* in = &incoming_[ir.info.round];
-
+            // ASCENT EDIT (queue_policy_ no longer a pointer)//
             bool unload = ((ir.info.round == exchange_round_) ? (block(lid(ir.info.to)) == 0) : (limit_ != -1))
-                          && queue_policy_->unload_incoming(*this, ir.info.from, ir.info.to, ir.message.size());
+                          && queue_policy_.unload_incoming(*this, ir.info.from, ir.info.to, ir.message.size());
 
             ir.place(in, unload, storage_, iexchange);
             ir.reset();

--- a/src/libs/apcomp/internal/diy/include/diy/master.hpp
+++ b/src/libs/apcomp/internal/diy/include/diy/master.hpp
@@ -86,7 +86,7 @@ namespace apcompdiy
       struct QueuePolicy
       {
                 QueuePolicy(size_t sz): size(sz)          {}
-                QueuePolicy(const QueuePolicy &p) size(p.size) {}
+                QueuePolicy(const QueuePolicy &p): size(p.size) {}
                ~QueuePolicy() {}
         bool    unload_incoming(const Master&, int, int, size_t sz) const         { return sz > size; }
         bool    unload_outgoing(const Master& master, int from, size_t sz) const  { return sz > size*master.outgoing_count(from); }

--- a/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp
+++ b/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp
@@ -85,6 +85,8 @@ namespace vtkhdiy
       struct QueuePolicy
       {
                 QueuePolicy(size_t sz): size(sz)          {}
+                QueuePolicy(const QueuePolicy &p) size(p.size) {}
+               ~QueuePolicy() {}
         bool    unload_incoming(const Master& master, int from, int to, size_t sz) const    { return sz > size; }
         bool    unload_outgoing(const Master& master, int from, size_t sz) const            { return sz > size*master.outgoing_count(from); }
 

--- a/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp
+++ b/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp
@@ -73,22 +73,24 @@ namespace vtkhdiy
       template<class Block>
       using Callback = std::function<void(Block*, const ProxyWithLink&)>;
 
-      struct QueuePolicy
-      {
-        virtual bool    unload_incoming(const Master& master, int from, int to, size_t size) const  =0;
-        virtual bool    unload_outgoing(const Master& master, int from, size_t size) const          =0;
-        virtual         ~QueuePolicy() {}
-      };
+      // BEGIN ASCENT CHANGE //
+      // struct QueuePolicy
+      // {
+      //   virtual bool    unload_incoming(const Master& master, int from, int to, size_t size) const  =0;
+      //   virtual bool    unload_outgoing(const Master& master, int from, size_t size) const          =0;
+      //   virtual         ~QueuePolicy() {}
+      // };
 
       //! Move queues out of core if their size exceeds a parameter given in the constructor
-      struct QueueSizePolicy: public QueuePolicy
+      struct QueuePolicy
       {
-                QueueSizePolicy(size_t sz): size(sz)          {}
+                QueuePolicy(size_t sz): size(sz)          {}
         bool    unload_incoming(const Master& master, int from, int to, size_t sz) const    { return sz > size; }
         bool    unload_outgoing(const Master& master, int from, size_t sz) const            { return sz > size*master.outgoing_count(from); }
 
         size_t  size;
       };
+      // END ASCENT CHANGE //
 
       struct MessageInfo
       {
@@ -173,7 +175,9 @@ namespace vtkhdiy
                            ExternalStorage*     storage  = 0,  //!< storage object (path, method, etc.) for storing temporary blocks being shuffled in/out of core
                            SaveBlock            save     = 0,  //!< block save function; master manages saving if save != 0
                            LoadBlock            load     = 0,  //!< block load function; master manages loading if load != 0
-                           QueuePolicy*         q_policy = new QueueSizePolicy(4096)): //!< policy for managing message queues specifies maximum size of message queues to keep in memory
+                           // BEGIN ASCENT EDIT //
+                           QueuePolicy          q_policy = QueuePolicy(4096)): //!< policy for managing message queues specifies maximum size of message queues to keep in memory
+                           // END ASCENT EDIT //
                       blocks_(create, destroy, storage, save, load),
                       queue_policy_(q_policy),
                       limit_(limit),
@@ -185,7 +189,12 @@ namespace vtkhdiy
                       exchange_round_(-1),
                       immediate_(true)
                                                         {}
-                    ~Master()                           { set_immediate(true); clear(); delete queue_policy_; }
+                    ~Master()                           { set_immediate(true);
+                                                          clear(); 
+                                                          // BEGIN ASCENT CHANGE
+                                                          //delete queue_policy_;
+                                                          // END ASCENT CHANGE
+                                                        }
       inline void   clear();
       inline void   destroy(int i)                      { if (blocks_.own()) blocks_.destroy(i); }
 
@@ -306,8 +315,10 @@ namespace vtkhdiy
       Collection            blocks_;
       std::vector<int>      gids_;
       std::map<int, int>    lids_;
-
-      QueuePolicy*          queue_policy_;
+      
+      // BEGIN ASCENT CHANGE
+      QueuePolicy           queue_policy_;
+      // END ASCENT CHANGE
 
       int                   limit_;
       int                   threads_;
@@ -528,7 +539,8 @@ unload_incoming(int gid)
     for (InQueueRecords::iterator it = in_qrs.records.begin(); it != in_qrs.records.end(); ++it)
     {
       QueueRecord& qr = it->second;
-      if (queue_policy_->unload_incoming(*this, it->first, gid, qr.size))
+      // ASCENT EDIT (queue_policy_ no longer a pointer)//
+      if (queue_policy_.unload_incoming(*this, it->first, gid, qr.size))
       {
         log->debug("Unloading queue: {} <- {}", gid, it->first);
         qr.external = storage_->put(in_qrs.queues[it->first]);
@@ -555,7 +567,8 @@ unload_outgoing(int gid)
     out_queues_size += it->second.size();   // buffer contents
     ++count;
   }
-  if (queue_policy_->unload_outgoing(*this, gid, out_queues_size - sizeof(size_t)))
+  // ASCENT EDIT (queue_policy_ no longer a pointer)//
+  if (queue_policy_.unload_outgoing(*this, gid, out_queues_size - sizeof(size_t)))
   {
       log->debug("Unloading outgoing queues: {} -> ...; size = {}\n", gid, out_queues_size);
       MemoryBuffer  bb;     bb.reserve(out_queues_size);
@@ -566,7 +579,8 @@ unload_outgoing(int gid)
         if (it->first.proc == comm_.rank())
         {
           // treat as incoming
-          if (queue_policy_->unload_incoming(*this, gid, it->first.gid, it->second.size()))
+          // ASCENT EDIT (queue_policy_ no longer a pointer)//
+          if (queue_policy_.unload_incoming(*this, gid, it->first.gid, it->second.size()))
           {
             QueueRecord& qr = out_qr.external_local[it->first];
             qr.size = it->second.size();
@@ -920,7 +934,8 @@ comm_exchange(ToSendList& to_send, int out_queues_limit)
           log->debug("Unloading outgoing directly as incoming: {} <- {}", to, from);
           MemoryBuffer& bb = it->second;
           in_qr.size = bb.size();
-          if (queue_policy_->unload_incoming(*this, from, to, in_qr.size))
+          // ASCENT EDIT (queue_policy_ no longer a pointer)//
+          if (queue_policy_.unload_incoming(*this, from, to, in_qr.size))
             in_qr.external = storage_->put(bb);
           else
           {
@@ -1040,8 +1055,9 @@ comm_exchange(ToSendList& to_send, int out_queues_limit)
       assert(ir.info.round >= exchange_round_);
       IncomingRound *in = &incoming_[ir.info.round];
 
+      // ASCENT EDIT (queue_policy_ no longer a pointer)//
       bool unload_queue = ((ir.info.round == exchange_round_) ? (block(lid(to)) == 0) : (limit_ != -1)) &&
-                          queue_policy_->unload_incoming(*this, from, to, size);
+                          queue_policy_.unload_incoming(*this, from, to, size);
       if (unload_queue)
       {
         log->debug("Directly unloading queue {} <- {}", to, from);

--- a/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp
+++ b/src/libs/vtkh/compositing/internal/diy/include/diy/master.hpp
@@ -85,7 +85,7 @@ namespace vtkhdiy
       struct QueuePolicy
       {
                 QueuePolicy(size_t sz): size(sz)          {}
-                QueuePolicy(const QueuePolicy &p) size(p.size) {}
+                QueuePolicy(const QueuePolicy &p): size(p.size) {}
                ~QueuePolicy() {}
         bool    unload_incoming(const Master& master, int from, int to, size_t sz) const    { return sz > size; }
         bool    unload_outgoing(const Master& master, int from, size_t sz) const            { return sz > size*master.outgoing_count(from); }

